source/grid/grid_tools.cc

```diff
@@ -1595,6 +1595,37 @@ namespace GridTools
 
 
   template <int dim, int spacedim>
+  std::vector<std::set<typename Triangulation<dim,spacedim>::active_cell_iterator> >
+  vertex_to_cell_map(const Triangulation<dim,spacedim> &triangulation)
+  {
+    std::vector<std::set<typename Triangulation<dim,spacedim>::active_cell_iterator> >
+    vertex_to_cell_map(triangulation.n_vertices());
+    typename Triangulation<dim,spacedim>::active_cell_iterator cell = triangulation.begin_active(),
+                                                               endc = triangulation.end();
+    for (; cell!=endc; ++cell)
+      for (unsigned int i=0; i<GeometryInfo<dim>::vertices_per_cell; ++i)
+        vertex_to_cell_map[cell->vertex_index(i)].insert(cell);
+
+    // Take care of hanging nodes
+    cell = triangulation.begin_active();
+    for (; cell!=endc; ++cell)
+      for (unsigned int i=0; i<GeometryInfo<dim>::faces_per_cell; ++i)
+        {
+          if ((cell->neighbor_level(i)>=0) && (cell->neighbor_level(i)<cell->level()))
+            {
+              typename Triangulation<dim,spacedim>::active_cell_iterator adjacent_cell =
+                cell->neighbor(i);
+              for (unsigned int j=0; j<GeometryInfo<dim>::vertices_per_face; ++j)
+                vertex_to_cell_map[cell->face(i)->vertex_index(j)].insert(adjacent_cell);
```

Comment:
Think of a situation where four cells come together at one edge, and one of them is refined. Then you add the two neighbors of the cell as adjacent to the central vertex of the (refined) edge. But you forget to add the cell that is "across the edge" from the refined cell because it is not a (coarser) neighbor of the child cells.